### PR TITLE
Document the CI setup

### PR DESCRIPTION
**Pull request by dave-tucker**
This documents the current CI setup and how to run and write tests.

Signed-off-by: Dave Tucker <dt@docker.com>

### CODE REVIEW

File: docs/ci.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+LinuxKit CI
+===========
+
+The LiunxKit CI system is uses [DatakitCI](https://github.com/moby/datakit/tree/master/ci)
```
Comment (dave-tucker) [inline docs/ci.md]: `s/is uses/uses/g`

File: docs/ci.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+The configuration can be found [here](https://github.com/linuxkit/linuxkit-ci)
+
+## PR Testing
+
+Each PR is tested on disposable VM's spawned in Google Cloud Platform
```
Comment (dave-tucker) [inline docs/ci.md]: `on a disposable VM`

File: docs/ci.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+If the test passed then we will run a set of tests to check that LinuxKit images are able to be run on all supported platforms.
+These are run on the LinuxKitCI controller inside a Docker Container.
+
+LinuxKit CI will run `make ci-platforms`
+Results will be collected from `test/_results` and added to DataKit
+
+## Branch and Tag testing
+
+Branches and Tags are tested on disposable VM's spawned in GCP.
```
Comment (dave-tucker) [inline docs/ci.md]: on disposable VMs

File: docs/ci.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+The LiunxKit CI system is uses [DatakitCI](https://github.com/moby/datakit/tree/master/ci)
+The configuration can be found [here](https://github.com/linuxkit/linuxkit-ci)
+
+## PR Testing
+
+Each PR is tested on disposable VM's spawned in Google Cloud Platform
+This machine has no privileges or credentials to talk to GCP or other cloud platforms.
+
+FIXME: Add instructions on how to build a base image for LinuxKitCI and GCP
```
Comment (dave-tucker) [inline docs/ci.md]: @talex5 I remember we discussed this briefly but don't recall reaching a conclusion.
Using `moby` to build the CI images would be awesome.

File: docs/ci.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+LinuxKit CI runs `make ci-pr` in the VM
+This will run the tests using `rtf` and the results directory is SCP'd back to the controller.
+The test results will be stored in DataKit for additional access
```
Comment (dave-tucker) [inline docs/ci.md]: `s/additional access//g`

**Comment by rneugeba** [comment]
Having documentation is good...checking in know inaccurate documentation not so...

I would also prefer if there is *one* document, maybe `docs/testing.md` which describes both the CI setup and how devs can run tests (and write tests)

**Comment by dave-tucker** [comment]
Ok, let me try this another way...
I'll document how it's set up and how it works today.
Then I'll open a PR with suggested improvements, which wouldn't be merged until such times as the improvements are implemented...

**Comment by dave-tucker** [comment]
Done.

File: docs/testing.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+Branches and Tags are tested on a dedicated machine that runs in GCP.
+
+LinuxKit CI runs `make ci` or `make ci-tag` in the VM
+This target runs the tests using `rtf` and the results directory is SCP'd back to the controller.
+The test results will be stored in DataKit.
+
+If the tests pass, the GCP test is run in the same manner as described for PR tests.
+
+Finally, CI will run `make test-ltp` which will run the Linux Testing Project tests.
```
Comment (justincormack) [inline docs/testing.md]: It would be good to link to the code which does this

Comment (dave-tucker) [inline docs/testing.md]: The link to linuxkit-ci repo is at the top of the section. 
I'm not sure linking directly is a good idea as it's likely to change...

**Pull request merged**